graylog2-web-interface/src/views/components/widgets/events/EventsList/EventsList.test.tsx

```diff
@@ -0,0 +1,233 @@
+import TestStoreProvider from 'views/test/TestStoreProvider';
+import { loadViewsPlugin, unloadViewsPlugin } from 'views/test/testViewsPlugin';
+import useAutoRefresh from 'views/hooks/useAutoRefresh';
+import EventsWidgetConfig from 'views/logic/widgets/events/EventsWidgetConfig';
+
+import EventsList from './EventsList';
+
+jest.mock('views/hooks/useAutoRefresh');
+
+const dummySearchJobResults = {
+  errors: [],
+  execution: { cancelled: false, completed_exceptionally: false, done: true },
+  id: 'foo',
+  owner: 'me',
+  search_id: 'bar',
+  results: {},
+};
+jest.mock('views/components/widgets/reexecuteSearchTypes');
+jest.mock('stores/useAppDispatch');
+
+describe('EventsList', () => {
+  const config = EventsWidgetConfig.createDefault();
+
+  const data = {
+    id: 'search-type-id',
+    type: 'events' as const,
+    events: [
+      {
+        id: '01HV0YS4GHDMT30E3EMWQVQNK9',
+        streams: [
+          '000000000000000000000001',
+        ],
+        timestamp: '2024-04-09T08:14:49.445Z',
+        message: 'Test Event 1',
+        alert: false,
+        event_definition_id: 'event-definition-1',
+        priority: 2,
+        event_keys: [],
+        replay_info: {
+          timerange_start: '2024-04-09T08:14:46.644Z',
+          timerange_end: '2024-04-09T08:14:49.644Z',
+          query: '',
+          streams: [
+            '000000000000000000000001',
+          ],
+          filters: [],
+        },
+        status: 'NEW',
+        opened_at: '2024-04-09T08:14:49.445Z',
+        closed_at: null,
+        updated_at: null,
+        owner: null,
+        risk_score: 2,
+        archived: false,
+        assigned_to: undefined,
+        created_at: '2024-04-09T08:14:49.445Z',
+        name: undefined,
+      },
+      {
+        id: '01HV0YS4GH0VC7DV6A2VGN1VJ0',
+        streams: [
+          '000000000000000000000001',
+        ],
+        timestamp: '2024-04-09T08:14:49.416Z',
+        message: 'Test Event 2',
+        alert: false,
+        event_definition_id: 'event-definition-2',
+        priority: 2,
+        event_keys: [],
+        replay_info: {
+          timerange_start: '2024-04-09T08:14:46.644Z',
+          timerange_end: '2024-04-09T08:14:49.644Z',
+          query: '',
+          streams: [
+            '000000000000000000000001',
+          ],
+          filters: [],
+        },
+        status: 'NEW',
+        opened_at: '2024-04-09T08:14:49.416Z',
+        closed_at: null,
+        updated_at: null,
+        owner: null,
+        risk_score: 2,
+        archived: false,
+        assigned_to: undefined,
+        created_at: '2024-04-09T08:14:49.445Z',
+        name: undefined,
+      },
+    ],
+    totalResults: 1,
+  };
+
+  beforeAll(() => {
+    loadViewsPlugin();
+
+    asMock(useAutoRefresh).mockReturnValue({
+      refreshConfig: null,
+      startAutoRefresh: () => {},
+      stopAutoRefresh: () => {},
+    });
+  });
+
+  afterAll(unloadViewsPlugin);
```

Comment:
You can also use `usePlugin` to replace loading the plugin in `beforeAll`/unloading `afterAll`.